api/etcd/watch.go

```diff
@@ -12,43 +13,65 @@ import (
 	"github.com/pkg/errors"
 )
 
-// Watch the given prefix for anything new.
-func (s *RulesetService) Watch(ctx context.Context, prefix string, revision string) (*api.RulesetEvents, error) {
+// Watch a list of paths for changes and return a list of events. If the list is empty or nil,
+// watch all paths. If the revision is empty, watch from the latest revision.
+// The given context can be used to limit the watch period or to cancel any running one.
+func (s *RulesetService) Watch(ctx context.Context, paths []string, revision string) (*api.RulesetEvents, error) {
 	ctx, cancel := context.WithCancel(ctx)
 	defer cancel()
 
 	opts := []clientv3.OpOption{clientv3.WithPrefix()}
-	if i, _ := strconv.ParseInt(revision, 10, 64); i > 0 {
+	rev, _ := strconv.ParseInt(revision, 10, 64)
+	if rev > 0 {
 		// watch from the next revision
-		opts = append(opts, clientv3.WithRev(i+1))
+		opts = append(opts, clientv3.WithRev(rev+1))
 	}
 
-	events := api.RulesetEvents{
-		Revision: revision,
-	}
+	var events api.RulesetEvents
 
-	wc := s.Client.Watch(ctx, s.rulesPath(prefix, ""), opts...)
+	wc := s.Client.Watch(ctx, s.rulesPath("", ""), opts...)
```

Comment:
Instead of watching the whole tree, I found out the [WithFromKey](https://godoc.org/github.com/coreos/etcd/clientv3#WithFromKey) option  maybe you can use it to specify the keys to watch?

Reply:
If it's doable it will reduce the complexity of this function a lot.

Reply:
Unfortunately, it only works for one key

Reply:
Can't you chain the call?

Reply:
Not for watches

Reply:
I mean `s.Client.Watch(ctx, s.rulespath(...), WithFromKey(k1), WithFromKey(k2),...)`.

Reply:
`WithFromKey` doesn't take any argument, it uses the key (the second argument of `Watch`) as the key to watch